### ISSUE TITLE
Ms Excel - Shared Runtime OfficeJs exception

### ISSUE THREAD

**Issue by ishanig11** [OfficeDev/office-js#4255]
We are trying to develop an Office Addin using OfficeJs APIs for Ms Excel.
While hitting   Office API, we are getting this error

![image](https://github.com/OfficeDev/office-js/assets/157671419/b747dae8-5e69-4ebc-82d7-e86e11629e5c)

Then we referred to the official documentation to include Shared Runtime in our Manifest and Project.
[Configure your Office Add-in to use a shared runtime - Office Add-ins | Microsoft Learn](https://learn.microsoft.com/en-us/office/dev/add-ins/develop/configure-your-add-in-to-use-a-shared-runtime)

We followed exact steps and updated both our Manifest and Client code.
But nothing worked.
Still, it gives the same error.
Let us know if we have missed anything which is not documented in the above link. 
It is a blocker for us and needs to be resolved asap.

Thanks in advance

**Comment by zhenhuangMSFT** [comment]
Hi @ishanig11, thanks for raising the issue! Did you hit this at the very beginning or not until recently?

**Comment by ishanig11** [comment]
We started development recently with Excel Addin and faced this problem in the beginning.

**Comment by ishanig11** [comment]
Any update? This is blocker issue for our development.

**Comment by mhelles** [comment]
@ishanig11 can you please provide a manifest for your add in so we can debug?

**Comment by ishanig11** [comment]
[manifest.zip](https://github.com/OfficeDev/office-js/files/14761742/manifest.zip)
PFA test manifest file. You can check setting <Runtime resid="Taskpane.New.Url" lifetime="long" />

**Comment by ishanig11** [comment]
@mhelles  @zhenhuangMSFT  Any update? This is blocker issue now.

**Comment by mhelles** [comment]
So I took a look at your manifest and found a few problems. Based on the documentation you linked above,  the Runtimes section should be immediately after the opening Host tag. Also, your functionfile is not pointing to the "Taskpane.New.Url" in the FunctionFile tag. 

Note steps 4 and 6 here: https://learn.microsoft.com/en-us/office/dev/add-ins/develop/configure-your-add-in-to-use-a-shared-runtime

Let me know if those changes don't work.

**Comment by ishanig11** [comment]
@mhelles  : Attached manifest file. I have made changes as per suggestion. Now getting error as 
Invalid manifest from addincommands dev catalog. Exception details: {"name":"Sys.ArgumentException","paramName":"Manifest parser failure: resid and taskpaneid not correct"}

[manifest 3.zip](https://github.com/OfficeDev/office-js/files/14832186/manifest.3.zip)

**Comment by ishanig11** [comment]
@mhelles : Did you get chance to check newly updated manifest? Please check as its blocker issue.

**Comment by mhelles** [comment]
What tools have you used to debug your manifest? Here is a link to how to debug the manifest file, that should be able to help you determine what resid and taskpaneid are failing: https://learn.microsoft.com/en-us/office/dev/add-ins/testing/troubleshoot-manifest

Please note, we are not able to always offer same day responses here on Github.

**Comment by ishanig11** [comment]
@mhelles 
[ManifestValidation.txt](https://github.com/OfficeDev/office-js/files/14883174/ManifestValidation.txt)
I have already executed it. And as per that its valid. please refer attached validation logs.

I am importing manifest file using Add-in command of Word in browser.

**Comment by ishanig11** [comment]
@mhelles  Any update?

**Comment by mhelles** [comment]
I just took a few mins to take a look at your manifest and you use ButtonId2 twice, you will need distinct IDs for each button. I changed it to ButtonId02 for your Document Info button and was able to load your manifest. 

Please try that.

**Comment by ishanig11** [comment]
Thank you @mhelles , it worked if you have single URL.
But how to handle if you are calling different URLs for taskpne.
[manifest 4.zip](https://github.com/OfficeDev/office-js/files/14996182/manifest.4.zip)

Refer above manifest file. URL formats like https://localhost:3000/taskpane.html?scope=new , https://localhost:3000/taskpane.html?scope=sessions etc

Thanks

**Comment by mhelles** [comment]
I'm not really sure what you are asking, you have different URLs in your manifest and buttons are associated with an action with different sourceLocations. This page should cover your questions: https://learn.microsoft.com/en-us/office/dev/add-ins/develop/xml-manifest-overview?tabs=tabid-1

**Comment by ishanig11** [comment]
@mhelles : Basically I am looking to implement shared runtime for multiple task pane.
As show/hide task pane API is not working in standalone mode.
![image](https://github.com/OfficeDev/office-js/assets/157671419/b32c0568-2274-4463-8c81-3b4909dc8996)
Let me know if there is any other way to make it work.

**Comment by m-hellesen** [comment]
Hey there, from the following link: "Don't design your add-in to use multiple task panes if you are planning to use a shared runtime. A shared runtime only supports the use of one task pane. Note that any task pane without a <TaskpaneID> is considered a different task pane." 
https://learn.microsoft.com/en-us/office/dev/add-ins/develop/configure-your-add-in-to-use-a-shared-runtime

**Comment by ishanig11** [comment]
Yes, I know it. But then to build my functionality, I need to know which button is clicked from Ribbon inside my taskpane code.
Is it possible to get if user clicks on which button from custom menu of Ribbon. Like if user clicks on  New/Open/Save etc button then I want to show respective HTML in taskpane.

Can you please tell me way to achieve this?

Thanks

**Comment by amanism19** [comment]
✅ Recommended Pattern: Pass Context from Ribbon → Task Pane
Instead of creating multiple task panes, you:

Keep one task pane
Use different ribbon buttons
Store which button was clicked in shared state
Let the task pane render different HTML based on that state

This approach is explicitly enabled by shared runtime design, where:

“any buttons that your add-in displays on the ribbon will run in the same shared runtime” https://learn.microsoft.com/en-us/office/dev/add-ins/develop/configure-your-add-in-to-use-a-shared-runtime?tabs=jsonmanifest
